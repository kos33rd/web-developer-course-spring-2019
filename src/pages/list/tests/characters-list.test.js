import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import createStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import CharactersList from '../characters-list'

describe('CharactersList', () => {
    it('test render empty component', () => {
        const mockStore = createStore([thunk])
        const initialState = {
            app: {
                characters: []
            }
        }
        const store = mockStore(initialState)


        const container = renderer
            .create(
                <Provider store={store}>
                    <Router>
                        <Route component={CharactersList}/>
                    </Router>
                </Provider>,
                { createNodeMock: ({ type }) => document.createElement(type) }
            )

        // const instance = container.instance()
        expect(container.toJSON()).toMatchSnapshot()
    })

})
