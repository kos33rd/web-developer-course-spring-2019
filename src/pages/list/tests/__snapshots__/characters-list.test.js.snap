// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersList test render empty component 1`] = `
<div
  className="body"
>
  <h3
    className="title"
  >
    Now with 
    <span
      className="rainbow"
    >
      flexbox
    </span>
  </h3>
  <div
    className="container row"
  />
</div>
`;
